### PR TITLE
Migration: Data/Panel link editor

### PR DESCRIPTION
**Pull request by tskarhed**
#22862
We might need some kind of separator for the Data links, as it is a bit unclear where one begins and the other ends.
Screenshot from 2020-04-22 12-27-42
Screenshot from 2020-04-22 12-28-02

### CODE REVIEW

File: packages/grafana-ui/src/components/DataLinks/DataLinkEditor.tsx [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+import { Button } from '../Button';
 
 interface DataLinkEditorProps {
   index: number;
   isLast: boolean;
   value: DataLink;
   suggestions: VariableSuggestion[];
   onChange: (index: number, link: DataLink, callback?: () => void) => void;
-  onRemove: (link: DataLink) => void;
+  onRemove?: (link: DataLink) => void;
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/DataLinks/DataLinkEditor.tsx:17]: Will making this optional have any consequences?

**Comment by torkelo** [comment]
Dam think you misunderstood how the inline should work , it should not be a form. It should be unchanged. Only change the modal form

**Comment by tskarhed** [comment]
Okay, gotcha!

**Comment by tskarhed** [comment]
Now data links look like this in the Graph panel too!
Screenshot from 2020-04-22 14-04-48

**Comment by torkelo** [comment]
Can we move the switch toggle in the bottom in the modal form? Looks a bit strange in the middle between those two big inputs

**Comment by tskarhed** [comment]
I think I broke the syntax highlighting :disappointed: Going to have a look at it tomorrow

File: packages/grafana-ui/src/components/DataLinks/DataLinksEditor.tsx [hunk 2]
```diff
@@ -1,85 +0,0 @@
...
-  maxLinks?: number;
-}
-
-export const enableDatalinksPrismSyntax = () => {
-  Prism.languages['links'] = {
-    builtInVariable: {
-      pattern: /(\${\S+?})/,
-    },
-  };
-};
```
Comment (dprokop) [inline packages/grafana-ui/src/components/DataLinks/DataLinksEditor.tsx:26]: Removing this fn is the reasons why syntax highlighting is not working

File: packages/grafana-ui/src/components/DataLinks/DataLinksEditor.tsx [hunk 3]
```diff
@@ -1,85 +0,0 @@
...
-
-export const DataLinksEditor: FC<DataLinksEditorProps> = React.memo(
-  ({ value = [], onChange, suggestions, maxLinks }) => {
-    const theme = useTheme();
-    enableDatalinksPrismSyntax();
```
Comment (dprokop) [inline packages/grafana-ui/src/components/DataLinks/DataLinksEditor.tsx:31]: and this call is required for syntax highlighting to work

**Review by dprokop — APPROVED** [review]
Looks and works great! 

@aocenas can you please check if derived field links work as expected?

**Review by dprokop — CHANGES_REQUESTED** [review]
Some sizing issue in Elasticsearch data links:

image

Sorry for my bipolar approve-request changes :D

**Comment by tskarhed** [comment]
Fixed!
Screenshot from 2020-04-23 14-44-19

**Review by dprokop — APPROVED** [review]
Looks good now!

**Comment by aocenas** [comment]
@dprokop sorry for late response I see you already checked but yeah don't see any issue right now.

**Comment by dprokop** [comment]
@aocenas no worries, I've tested it too and it seems to work just fine

**Pull request merged**